junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/ForkJoinPoolHierarchicalTestExecutorService.java

```diff
@@ -51,7 +52,9 @@
 public class ForkJoinPoolHierarchicalTestExecutorService implements HierarchicalTestExecutorService {
 
 	private final ForkJoinPool forkJoinPool;
+	private final TaskEventListener taskEventListener;
 	private final int parallelism;
+	private final ThreadLocal<ThreadLock> threadLocks = ThreadLocal.withInitial(ThreadLock::new);
```

Comment:
Yes, it would. I think we could remove the `.remove()` call to further simplify it without changing the behavior. Any objections?

Reply:
Since we own the FJ pool and close it after the execution, we can get away with "leaking" the thread locals. If we used the common FJ pool, this would be a problem.

Reply:
I have decided to further simplify it in 5256049af3655c80599e7df839049f4272d16ffc